src/strategy/raids/moltencore/RaidMcActions.cpp

```diff
@@ -1,43 +1,220 @@
+    return false;
+}
+
+bool McShazzrahMoveAwayAction::Execute(Event event)
+{
+    if (Unit* boss = AI_VALUE2(Unit*, "find target", "shazzrah"))
     {
-        // add/remove from both for now as it will make it more obvious to
-        // player if this strat remains on after fight somehow
-        botAI->ChangeStrategy("-move from group", BOT_STATE_NON_COMBAT);
-        botAI->ChangeStrategy("-move from group", BOT_STATE_COMBAT);
-        return true;
+        float distToTravel = ARCANE_EXPLOSION_DISTANCE - bot->GetDistance(boss);
+        if (distToTravel > 0)
+            return MoveAway(boss, distToTravel);
     }
     return false;
 }
 
-bool McMoveFromBaronGeddonAction::Execute(Event event)
+bool McGolemaggMarkBossAction::Execute(Event event)
 {
-    const float radius = 25.0f;  // more than should be needed but bots keep trying to run back in
-    if (Unit* boss = AI_VALUE2(Unit*, "find target", "baron geddon"))
+    if (Unit* boss = AI_VALUE2(Unit*, "find target", "golemagg the incinerator"))
     {
-        long distToTravel = radius - bot->GetDistance(boss);
-        if (distToTravel > 0)
+        if (Group* group = bot->GetGroup())
         {
-            // float angle = bot->GetAngle(boss) + M_PI;
-            // return Move(angle, distToTravel);
-            return MoveAway(boss, distToTravel);
+            ObjectGuid currentSkullGuid = group->GetTargetIcon(RtiTargetValue::skullIndex);
+            if (currentSkullGuid.IsEmpty() || currentSkullGuid != boss->GetGUID())
+            {
+                group->SetTargetIcon(RtiTargetValue::skullIndex, bot->GetGUID(), boss->GetGUID());
+                return true;
+            }
         }
     }
     return false;
 }
+
+bool McGolemaggTankAction::MoveUnitToPosition(Unit* target, const Position& tankPosition, float maxDistance)
+{
+    if (bot->GetVictim() != target)
+        return Attack(target);
+    if (target->GetVictim() == bot)
+    {
+        float distanceToTankPosition = bot->GetExactDist2d(tankPosition.GetPositionX(), tankPosition.GetPositionY());
+        if (distanceToTankPosition > maxDistance)
+        {
+            float dX = tankPosition.GetPositionX() - bot->GetPositionX();
+            float dY = tankPosition.GetPositionY() - bot->GetPositionY();
+            float dist = sqrt(dX * dX + dY * dY);
+            float moveX = bot->GetPositionX() + (dX / dist) * maxDistance;
+            float moveY = bot->GetPositionY() + (dY / dist) * maxDistance;
+            return MoveTo(bot->GetMapId(), moveX, moveY, bot->GetPositionZ(), false, false,
+                          false, false, MovementPriority::MOVEMENT_COMBAT, true,
+                          true);
+        }
+
+        float orientation = atan2(target->GetPositionY() - bot->GetPositionY(),
+                                  target->GetPositionX() - bot->GetPositionX());
+        bot->SetFacingTo(orientation);
+    }
+    else if (!bot->IsWithinMeleeRange(target))
```

Comment:
Don't they do this on their own? They always run up the target they're attacking, is there a reason why you're manually handling this on top of that?